jwst/photom/photom.py

```diff
@@ -796,6 +803,12 @@ def photom_io(self, tabdata, order=None):
                     conversion, no_cal = self.create_2d_conversion(self.input.slits[self.slitnum],
                                                                    self.exptype, conversion,
                                                                    waves, relresps, order)
+            elif isinstance(self.input, datamodels.MultiSpecModel):
```

Comment:
There is a very remote but possible case that one would use NINT=1 in the SUBSTRIP96 subarray mode. In which case, only one spectrum would be extracted. In any case,  with NINT=1, extract-1d saves the spectra in a SpecModel and loops on spectral orders. We will probably need to test with simulated data.

Reply:
I reverted to old habits and checked extract_1d.extract instead of soss_extract for this behavior... extract_1d's output for the default algorithm is always a MultiSpecModel, with only one SpecModel appended to its `spec` list. From the lines Loïc pointed to, the same behavior is present in soss_extract - output_model is initialized as a MultiSpecModel, and results are appended to its `spec` list. Even if only one order of one image is processed, the output_model type does not change.